src/slang_frontend.cc

```diff
@@ -2684,6 +2685,120 @@ struct PopulateNetlist : public TimingPatternInterpretor, public ast::ASTVisitor
 		visitDefault(sym);
 	}
 
+	void handle(const ast::PrimitiveInstanceSymbol &sym)
+	{
+		auto ports = sym.getPortConnections();
+		std::string yosys_type(sym.primitiveType.name);
+		std::string yosys_name(sym.name);
+		bool inv_y = false;
+		bool inv_en = false;
+		RTLIL::Cell *cell;
+		auto &assign = ports.front()->as<ast::AssignmentExpression>();
```

Comment:
It'd be best to put `ast_invariant(sym, ports.front().kind == ast::ExpressionKind::Assignment);` above this to document the assumption